ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -555,18 +558,45 @@ func (h *FaultHandler) StopNetworkPacketLoss() func(http.ResponseWriter, *http.R
 		rwMu.Lock()
 		defer rwMu.Unlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Invoke the stop fault injection functionality if running
-
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("stopped")
-		logger.Info("Successfully stopped fault", logger.Fields{
+		var responseBody types.NetworkFaultInjectionResponse
+		var httpStatusCode int
+		stringToBeLogged := "Failed to stop fault"
+		// All command executions for the start network packet loss workflow all together should finish within 5 seconds.
+		// Thus, create the context here so that it can be shared by all os/exec calls.
+		ctx, cancel := context.WithTimeout(context.Background(), requestTimeoutDuration)
+		defer cancel()
+		// Check the status of current fault injection.
+		_, packetLossFaultExists, err := h.checkTCFault(ctx, taskMetadata)
+		if err != nil {
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(err.Error())
+			httpStatusCode = http.StatusInternalServerError
+		} else {
+			// If there doesn't already exist a network-packet-loss fault
+			if !packetLossFaultExists {
+				stringToBeLogged = "Successfully stopped fault"
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("stopped")
+				httpStatusCode = http.StatusOK
+			} else {
+				// Invoke the start fault injection functionality if not running.
+				err := h.stopNetworkPacketLossFault(ctx, taskMetadata, request)
+				if err != nil {
```

Comment:
nit: ditto here regarding whether we should check if the context has timed out

Reply:
Addressing this in the next revision